src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1146,10 +1147,12 @@ protected override void GetItem(string path)
                         }
                     }
                 }
+#endif
 
                 FileSystemInfo result = GetFileSystemItem(path, ref isContainer, false);
                 if (result != null)
                 {
+#if !UNIX
```

Comment:
This #if/#endif block is cumbersome to read.  Suggest instead:
if (result != null)
{
#if UNIX
     WriteItemObject(...)
#else
    original code
 }